src/components/calloutForm.tsx

```diff
@@ -0,0 +1,225 @@
+    border-bottom: 1px ${neutral[86]} solid;
+    position: relative;
+    margin: ${remSpace[4]} 0 ${remSpace[9]} 0;
+
+    h4 {
+        ${headline.xxxsmall({ fontWeight: 'bold' })};
+        margin: 0;
+    }
+
+    &:not([open]) .is-on, &[open] .is-off {
+        display: none;
+    }
+
+    .is-on, .is-off {
+        position: absolute;
+        transform: translate(0, 50%);
+        bottom: 0;
+    }
+
+    &[open] {
+        background: ${neutral[97]};
+    }
+
+    summary {
+        outline: none;
+        padding: 0;
+        list-style: none;
+        
+        &::-webkit-details-marker {
+            display: none;
+        }
+    
+        .campaign--kicker {
+            display: flex;
+            flex-direction: row;
+    
+            > .campaign--snippet__heading-logo {
+                flex: initial;
+    
+                .speech-bubble {
+                    padding: ${remSpace[3]};
+                    position: relative;
+                    color: ${neutral[100]};
+                    background-color: ${kicker};
+                    margin-bottom: ${remSpace[9]};
+                    min-width: 5.5rem;
+
+                    &::after {
+                        content: '';
+                        width: 1.25rem;
+                        height: 1.375rem;
+                        border-radius: 0 0 1.125rem;
+                        position: absolute;
+                        bottom: -0.75rem;
+                        left: 0.625rem;
+                        color: ${neutral[100]};
+                        background-color: ${kicker};
+                    }
+                }
+            }
+        }
+    }
+    
+    form {
+        margin: ${remSpace[4]} ${remSpace[2]} ${remSpace[9]} ${remSpace[2]};
+
+        a {
+            color: ${kicker};
+            text-decoration: none;
+            ${ textSans.small() };
+            position: absolute;
+            bottom: ${remSpace[2]};
+            right: ${remSpace[2]};
+        }
+
+        button {
+            margin: ${remSpace[4]} 0;
+        }
+
+        input {
+            margin-bottom: ${remSpace[4]};
+        }
+
+        p {
+            ${textSans.small()};
+        }
+    }
+
+    .description {
+        margin: ${remSpace[3]};
+
+        p {
+            ${body.small({ lineHeight: 'tight' })};
+            margin: ${remSpace[2]} 0;
+        }
+    }
+
+    .error-message {
+        color: ${text.error};
+    }
+
+    ${darkModeCss`
+        background: white;
+        color: ${neutral[7]};
+        border: none;
+    `}
+`;
+
+const optionalLabelStyles = css`
+    ${textSans.small({ fontStyle: "italic" })};
+    color: ${neutral[46]};
+`;
+
+const labelStyles = css`
+    ${textSans.medium({ fontWeight: 'bold' })};
+`;
+
+const renderField = ({ type, label, mandatory, options, id }: FormField): JSX.Element => {
+    switch (type) {
+        case 'text': {
+            return <TextInput name={`field_${id}`} label={label} optional={!mandatory}/>
```

Comment:
Could we store this field id in a `const` so we don't need to repeat the template string?

```tsx
const fieldId = `field_${id}`;
```